source/read-write-pref.txt

```diff
@@ -0,0 +1,228 @@
+       specified tags.
+
+   * - ``wtimeoutMS`` *(optional)*
+     - integer
+     - Specifies a time limit to prevent write operations from blocking indefinitely.
+
+   * - ``journal`` *(optional)*
+     - boolean
+     - Requests acknowledgment that the write operation has been written to the 
+       on-disk journal.
+
+Example: Set the Write Concern for a Single Write Operation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This code uses custom ``write_concern`` settings while creating new a document:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-write-concern
+   :end-before: end-write-concern
+
+Example: Retrieve and Apply an Existing Write Concern
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This code uses the ``new_write_concern`` method to construct a ``write_concern`` from
+the options of an existing database reference, ``myDB``. ``myDB`` can be 
+replaced with a reference to any entity that accepts a write concern option. 
+Then the new write concern is applied to a document, ``myDoc``:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-write-concern-2
+   :end-before: end-write-concern-2
+
+Read Concern
+------------
+
+The read concern specifies the following behaviors:
+
+- Level of :manual:`causal consistency </core/causal-consistency-read-write-concerns/>`
+  across replica sets.
+
+- `Isolation guarantees <https://www.mongodb.com/docs/current/core/read-isolation-consistency-recency/>`__ 
+  maintained during a query.
+
+You can specify the read concern setting by using the ``level`` parameter. The default
+read concern level is ``local``. This means that the client returns the data from the
+replica set member that the client is connected to, with no guarantee that the data has
+been written to all replica set members.
+
+.. note::
+    Lower read concern level requirements may reduce latency. 
+
+For more information about read concerns or read concern levels, see 
+:manual:`Read Concern </reference/read-concern/>` in the Server manual. For more
+detail on the ``read_concern`` type and definitions of the read concern levels, see
+the `Read Concern <https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Client.html#read_concern-instance_method>`__ 
+in the API documentation.
+
+Example: Set the Read Concern Level of an Aggregation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This code sets the read concern level of an aggregation to `"available"`:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-read-concern
+   :end-before: end-read-concern
+
+For more information about aggregates, see the `ruby-aggregation <https://www.mongodb.com/docs/ruby-driver/upcoming/reference/aggregation/>`_ 
+page.
```

Comment:
This should be a :ref: tag to the Aggregation page rather than a hardcoded link (and if one doesn't exist in the standardization branch yet then this can just be marked as a Todo for now)

Reply:
Bumping this comment since it's still a hard-coded link